lib/engine-dylib/src/artifact.rs

```diff
@@ -773,4 +794,41 @@ impl Artifact for DylibArtifact {
     fn serialize(&self) -> Result<Vec<u8>, SerializeError> {
         Ok(std::fs::read(&self.dylib_path)?)
     }
+
+    /// Serialize a `DylibArtifact` to a portable file
+    #[cfg(feature = "compiler")]
+    fn serialize_to_file(&self, path: &Path) -> Result<(), SerializeError> {
+        let serialized = self.serialize()?;
+        std::fs::write(&path, serialized)?;
+
+        /*
+        When you write the artifact to a new file it still has the 'Mach-O Identifier'
+        of the original file, and so this can causes linker issues when adding
+        the new file to an XCode project.
+
+        The below code renames the ID of the file so that it references itself through
+        an @executable_path prefix. Basically it tells XCode to find this file
+        inside of the projects' list of 'linked executables'.
+
+        You need to be running MacOS for the following to actually work though.
+        */
+        if path.extension().unwrap() == "dylib" {
+            let filename = path.file_name().unwrap().to_str().unwrap();
+            let parent_dir = path.parent().unwrap();
+            let absolute_path = std::fs::canonicalize(&parent_dir)
+                .unwrap()
+                .into_os_string()
+                .into_string()
+                .unwrap();
+
+            Command::new("install_name_tool")
+                .arg("-id")
+                .arg(format!("@executable_path/{}", &filename))
+                .arg(&filename)
+                .current_dir(&absolute_path)
+                .output()?;
```

Comment:
This only works when serializing to a file, but I assume if you change the location of that dylib file, it will stop working?
If the answer is no, then we should probably run this command after the linker has done their job creating the `dylib` (the `Artifact::new` function).

Reply:
This command will stop working if you rename the file, Because we prefix it with  `@executable_path/{file_name}`, It doesn't matter where it goes because it is just referencing itself of the same name inside the compiler source.

Unfortunately  this wouldn't worked inside `Artifact::new` because it has that tempfile name that will inevitably change.